src/core/model/Model.hpp

```diff
@@ -0,0 +1,23 @@
+#include "Model.h"
+
+namespace SEMBA {
+namespace Model {
+
+template<typename M>
+Model<M>::Model(
+	const M& mesh,
+	const PMGroup& physicalModels
+) : mesh(mesh),
+	physicalModels(physicalModels)
+{}
+
+template<typename M>
+Model<M>& Model<M>::operator=(const Model& rhs) {
+	mesh = rhs.mesh;
+	physicalModels = rhs.physicalModels;
+
+	return *this;

```

Comment:
I think this is potentially wrong because the new mesh needs to reassign the pointers to the physical models. I would add a test to reveal this.

Reply:
Replicated in tests. Good catch. Thanks!